tuist/tuist
# Fix generating CoreData models in resourceSynthesizer .coreData()

Resolves https://github.com/tuist/tuist/issues/6200

### Short description 📝

I have realized that during project mapping, resources and CoreData models are explicitly separated in the project structure (﻿`Target.resources` and `﻿Target.coreDataModels`). However, in the `﻿SynthesizedResourceInterfaceProjectMapper` class, when generating code from templates, the search for files is done only in ﻿`Target.resources`. When generating the CoreData template, it looks for files with the ﻿`xcdatamodeld` extension, which are located in `﻿Target.coreDataModels`.

- I have modified it to search for the appropriate files for code generation in both `resources` and `coreDataModels`.
- I have also enhanced the tests in ﻿`SynthesizedResourceInterfaceProjectMapperTests`, and running these tests before my changes will result in failures.

### How to test the changes locally 🧐

1. Create a project with `.coreData()` resourceSynthesizer
2. Run the `tuist` target with the parameters (`generate`, `--no-open`, `--path /path/to/project`)
3. Make sure that the `TuistCoreData+AppCore.swift` file appears in the tuist Derived folder

### Contributor checklist ✅


### Reviewer checklist ✅
